From cad0db51630df45378213bfcc33a236e930e0776 Mon Sep 17 00:00:00 2001
From: Cedric Bail <cedric.bail@free.fr>
Date: Sat, 11 Jan 2014 21:36:47 +0900
Subject: [PATCH] eina: fix memory leak of inserting the wrong value in the
 hash.

We where inserting the pointer data instead of the pointer, leading to
unaligned access on Sparc (Thanks Lutin to report it and Debian tools/infra
to help us catch it) and also a memory leak.
---
 src/lib/eina/eina_file_common.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/lib/eina/eina_file_common.c b/src/lib/eina/eina_file_common.c
index f6ae3be..d12f6c9 100644
--- a/src/lib/eina/eina_file_common.c
+++ b/src/lib/eina/eina_file_common.c
@@ -192,7 +192,7 @@ eina_file_virtual_map_new(Eina_File *file,
         map->refcount = 0;
 
         eina_hash_add(file->map, &key, map);
-        eina_hash_direct_add(file->rmap, map->map, map);
+        eina_hash_direct_add(file->rmap, &map->map, map);
      }
 
    map->refcount++;
-- 
1.8.5.2

